File: 1331-rank-transform-of-an-array/1331-rank-transform-of-an-array.cpp
class Solution {
public:
    vector<int> arrayRankTransform(vector<int>& arr) {
        int rank = 1;
        vector<pair<int, int>> nums;
        for(int i = 0; i < arr.size(); i++)
        {
            nums.push_back({arr[i], i});
        }
        sort(nums.begin(), nums.end());
        for(int i = 0; i < nums.size(); i++)
        {
            if(i > 0 && nums[i].first > nums[i - 1].first)
            {
                rank++;
            }
            arr[nums[i].second] = rank;
        }
        return arr;
    }
};

File: 1331-rank-transform-of-an-array/1331-rank-transform-of-an-array.py
class Solution:
    def arrayRankTransform(self, arr: List[int]) -> List[int]:
        rank = 1
        nums = [(num, i) for i, num in enumerate(arr)]
        nums.sort()
        for i in range(len(nums)):
            if i > 0 and nums[i][0] > nums[i - 1][0]:
                rank += 1
            arr[nums[i][1]] = rank
        return arr